basyx.aasrepository/basyx.aasrepository-backend/src/main/java/org/eclipse/digitaltwin/basyx/aasrepository/SimpleAasRepositoryFactory.java

```diff
@@ -28,36 +28,39 @@
 import org.eclipse.digitaltwin.basyx.aasservice.AasServiceFactory;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Value;
-import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression;
 import org.springframework.stereotype.Component;
 
 /**
- * AasRepository factory returning an in-memory backend AasRepository
+ * CRUD AAS repository factory
+ * 
+ * @author mateusmolina
  * 
- * @author schnicke, kammognie
  */
-@ConditionalOnExpression("'${basyx.aasrepository.backend}'.equals('InMemory') or '${basyx.backend}'.equals('InMemory')")
 @Component
-public class InMemoryAasRepositoryFactory implements AasRepositoryFactory {
+public class SimpleAasRepositoryFactory implements AasRepositoryFactory {
+
+	AasBackendProvider aasBackendProvider;
 
-	private AasServiceFactory aasApiFactory;
-	
-	private String aasRepositoryName;
+	AasServiceFactory aasServiceFactory;
+
+	String aasRepositoryName = null;
 
 	@Autowired(required = false)
-	public InMemoryAasRepositoryFactory(AasServiceFactory aasApiFactory) {
-		this.aasApiFactory = aasApiFactory;
+	public SimpleAasRepositoryFactory(AasBackendProvider aasBackendProvider, AasServiceFactory aasServiceFactory) {
+		this.aasBackendProvider = aasBackendProvider;
+		this.aasServiceFactory = aasServiceFactory;
 	}
-	
+
 	@Autowired(required = false)
-	public InMemoryAasRepositoryFactory(AasServiceFactory aasApiFactory, @Value("${basyx.aasrepo.name:aas-repo}") String aasRepositoryName) {
-		this(aasApiFactory);
+	public SimpleAasRepositoryFactory(AasBackendProvider aasBackendProvider, AasServiceFactory aasServiceFactory, @Value("${basyx.aasrepo.name:aas-repo}") String aasRepositoryName) {
+		this.aasBackendProvider = aasBackendProvider;
+		this.aasServiceFactory = aasServiceFactory;
```

Comment:
Please delegate line 56 and 57 to utilize the first constructor.

this(aasBackendProvider, aasServiceFactory);